### PR TITLE
Local: Handle noalias.addrspace in copyMetadataForLoad

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/103939?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#103939** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/103939?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#103938** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/103938?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#102461** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102461?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#102462](https://github.com/llvm/llvm-project/pull/102462) <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102462?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#102599](https://github.com/llvm/llvm-project/pull/102599) <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102599?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by arsenm** [comment]
ping

File: llvm/test/Transforms/InstCombine/loadstore-metadata.ll [hunk 1]
```diff
@@ -177,7 +177,7 @@ define i32 @test_load_cast_combine_noalias_addrspace(ptr %ptr) {
 ; Ensure (cast (load (...))) -> (load (cast (...))) preserves TBAA.
```
Comment (frasercrmck) [inline llvm/test/Transforms/InstCombine/loadstore-metadata.ll:177]: Should this comment reference `noalias.addrspace` rather than TBAA?

**Review by frasercrmck — APPROVED** [review]
LGTM other than nit

File: llvm/test/Transforms/InstCombine/loadstore-metadata.ll (hunk 1 above)
Comment (arsenm) [inline llvm/test/Transforms/InstCombine/loadstore-metadata.ll:177]:
```suggestion
; Ensure (cast (load (...))) -> (load (cast (...))) preserves no alias.addrspace.
```

**Pull request merged**